examples/graphrnn_s.py

```diff
@@ -0,0 +1,362 @@
+        )
+        if sampling:
+            output_sequences, self.hidden = self.rnn(input_sequences,
+                                                     self.hidden)
+        else:
+            output_sequences, self.hidden = self.rnn(input_sequences)
+        # Unpack RNN output.
+        output_sequences, output_length = rnnutils.pad_packed_sequence(
+            output_sequences, batch_first=True)
+
+        # MLP to get adjacency vectors.
+        output_sequences = self.adjacency_mlp(output_sequences)
+
+        return self.mask_out_bits_after_length(output_sequences, input_length)
+
+    def sample(self, batch_size, device, max_num_nodes):
+        r"""Sample a batch of graph sequences. Assumes that learned/generated
+        graphs are connected, as in the paper.
+
+        Args:
+            batch_size (int): The number of graphs to sample.
+            device (torch.device): The device to run the sampling on.
+            max_num_nodes (int): The maximum number of nodes that a sampled
+                graph may contain.
+
+        :rtype Tensor:
+        """
+        input_sequence = torch.ones(batch_size, 1, self.adjacency_size,
+                                    device=device)  # SOS.
+        is_not_eos = torch.ones(batch_size, dtype=torch.long)
+
+        sequences = torch.zeros(batch_size, max_num_nodes, self.adjacency_size)
+        seq_lengths = torch.zeros(batch_size, dtype=torch.long)
+        # Id of the node whose adjacency vector will be sampled.
+        # Node 0 is not included.
+        node_id = 0
+        with torch.no_grad():
+            self.hidden = torch.zeros(self.num_layers, batch_size,
+                                      self.hidden_size, device=device)
+            while is_not_eos.any():
+                node_id += 1
+                if node_id == max_num_nodes:
+                    break
+
+                output_sequence_probs = self.forward(input_sequence,
+                                                     torch.ones(batch_size),
+                                                     sampling=True)
+                mask = torch.rand_like(output_sequence_probs)
+                output_sequence = torch.gt(output_sequence_probs, mask)
+
+                # Identify the EOS sequences.
+                is_not_eos *= output_sequence.any(dim=-1).squeeze().cpu()
+                seq_lengths += is_not_eos
+
+                sequences[:, node_id - 1] = output_sequence[:, 0]
+                input_sequence = output_sequence.float()
+
+        # Clean irrelevant bits and enforce creation of connected graph.
+        # Pack to 1 + seq_lengths to include empty sequences
+        self.mask_out_bits_after_length(sequences, seq_lengths + 1)
+        sequences = sequences.tril()
+
+        return sequences[:, :seq_lengths.max()], seq_lengths
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--plot", action="store_true",
+                    help="Whether to plot graphs.")
+args = parser.parse_args()
```

Comment:
maybe good to have a default or "parse_known_args" so the example can be run easily without arguments?

Reply:
With `argparse`, `--<...>` arguments are optional, so this example can already be run without arguments.